SignalServiceKit/src/Contacts/TSThread.m

```diff
@@ -97,12 +97,14 @@ - (void)removeAllThreadInteractionsWithTransaction:(YapDatabaseReadWriteTransact
     NSMutableArray<NSString *> *interactionIds = [NSMutableArray new];
     YapDatabaseViewTransaction *interactionsByThread = [transaction ext:TSMessageDatabaseViewExtensionName];
     OWSAssert(interactionsByThread);
-    [interactionsByThread
-        enumerateKeysInGroup:self.uniqueId
-                  usingBlock:^(
-                      NSString *_Nonnull collection, NSString *_Nonnull key, NSUInteger index, BOOL *_Nonnull stop) {
-                      [interactionIds addObject:key];
-                  }];
+    [interactionsByThread enumerateKeysInGroup:self.uniqueId
+                                    usingBlock:^(NSString *collection, NSString *key, NSUInteger index, BOOL *stop) {
+                                        if (key.length < 1) {
```

Comment:
I noticed that its the _key_ not the _deserialized object_ that is invalid (probably nil).  Which looks a lot like database corruption.

Reply:
ah, good catch. Specifically - seems like our view is out of sync with the backing collection. 

Should we trigger a rebuild?

Reply:
> Should we trigger a rebuild?

Done.